Sources/NIOCore/SingleStepByteToMessageDecoder.swift

```diff
@@ -273,7 +273,7 @@ public final class NIOSingleStepByteToMessageProcessor<Decoder: NIOSingleStepByt
         }
 
         // force unwrapping is safe because nil case is handled already and asserted above
-        if self._buffer!.readerIndex > 0, self.decoder.shouldReclaimBytes(buffer: self._buffer!) {
+        if let readerIndex = self._buffer?.readerIndex, readerIndex > 0, self.decoder.shouldReclaimBytes(buffer: self._buffer!) {
             self._buffer!.discardReadBytes()
```

Comment:
I don't think I can do this in all cases because the reclaim and discard bytes operations would then only act on the copy of the buffer. I think we should leave this as-is.